django/db/models/sql/query.py

```diff
@@ -1288,6 +1290,27 @@ def names_to_path(self, names, opts, allow_many=True, fail_on_missing=False):
             except FieldDoesNotExist:
                 if name in self.annotation_select:
                     field = self.annotation_select[name].output_field
+                elif pos == 0:
+                    # Deprecated Behavior: When the `default_related_name` was set
+                    # and we didn't pass `related_name` kwarg in Field ctor,
+                    # in Django 1.9 we still use model name as lookup string
+                    # but default_related_name.
+                    # The right way to query is using default_related_name as lookup string
+                    # See #26230 and the test case
+                    # `test_show_deprecated_message_when_model_name_in_queryset_lookup`
+                    # in tests/model_options/test_default_related_name.py
+                    # to get this deprecated use case
+                    for rel in opts.related_objects:
+                        if (name == rel.related_model._meta.model_name and
+                        rel.related_name == rel.related_model._meta.default_related_name):
+                            related_name = rel.related_name
+                            field = opts.get_field(related_name)
+                            warnings.warn(
+                                "Using model name %r in queryset lookup "
+                                "when meta.default_related_name %r is set is deprecated "
```

Comment:
missing period
the message could be: "Query lookup '%s' is deprecated in favor of Meta.default_related_name '%s'."